torch/distributed/_tensor/dispatch.py

```diff
@@ -230,6 +238,26 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
             redistribute_with_schema=needs_redistribute,
         )
 
+        aten = torch.ops.aten
+        random_ops = [
+            aten.native_dropout.default,
+            aten.normal_.default,
+            aten.uniform_.default,
+        ]
+        # before running local op computation, check if op is random op
+        # for random ops, set RNG offset
+        if op_call in random_ops:
+            assert isinstance(mesh, DeviceMesh)
+            if is_rng_supported_mesh(mesh):
+                dtensor_arg = arg_list[0]
+                assert isinstance(dtensor_arg, dtensor.DTensor)
+                old_offset = _get_rng_offset(mesh)
+                set_pre_op_offset(dtensor_arg._spec)
+            else:
+                warnings.warn(
+                    f"DTensor random operators may not have complete support on {mesh.device_type} device mesh"
```

Comment:
can we make this warning in the `is_rng_supported_mesh` directly?

Reply:
I don't think we should do it because `is_rng_supported_mesh` is only a condition check function. We should not raise warning in a bool function but places where failing the check may cause problems (e.g. the mesh must be a cuda device mesh to continue executing the following operations).

Reply:
I mean, it should not be a simple condition check, you should raise warning if device_type is not "cuda", and then return whether it's cuda or not. My point is the rng_supported_mesh should handle warnings too as that's why we need this function, otherwise you can just put the condition check in the if condition as it's just a simple check :)

Reply:
i see your point. let me put the warining in it.